pkg/storage/db.go

```diff
@@ -1060,3 +1083,83 @@ order by ae.ts asc
 	}
 	return nil
 }
+
+// StoreAccountOwner is an implementation of AccountOwnerStorer interface that saves account ID, its owner and champions of the account to a database
+func (db *DB) StoreAccountOwner(ctx context.Context, accountOwner domain.AccountOwner) error {
+
+	tx, err := db.sqldb.Begin()
+	if err != nil {
+		return err
+	}
+
+	err = db.storeAccountOwner(ctx, accountOwner, tx)
+
+	if err != nil {
+		if rollbackErr := tx.Rollback(); rollbackErr != nil {
+			return errors.Wrap(rollbackErr, err.Error()) // so we don't lose the original error
+		}
+		return err
+	}
+	if err = tx.Commit(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (db *DB) storeAccountOwner(ctx context.Context, accountOwner domain.AccountOwner, tx *sql.Tx) error {
+
+	sqlStatement := fmt.Sprintf(`REPLACE INTO aws_account (account) VALUES($1)`)
+	var err error
+	if _, err = tx.ExecContext(ctx, sqlStatement, accountOwner.AccountID); err != nil {
+		return err
+	}
+	// Insert or update details of account owner in "person" table
+	if _, err = tx.ExecContext(ctx, insertPersonQuery, accountOwner.Owner.Login, accountOwner.Owner.Email, accountOwner.Owner.Name, accountOwner.Owner.Valid); err != nil {
+		return err
+	}
+
+	sqlStatement = fmt.Sprintf(`SELECT id FROM person WHERE login=$1`)
+	row, err := tx.QueryContext(ctx, sqlStatement, accountOwner.Owner.Login)
+	if err != nil {
+		return err
+	}
+	var id int
+	row.Next()
+	if err := row.Scan(&id); err != nil {
+		return err
+	}
+
+	sqlStatement = fmt.Sprintf(insertAccountOwnerQuery, id)
+	if _, err := tx.ExecContext(ctx, sqlStatement, accountOwner.AccountID); err != nil {
```

Comment:
Question: Why does the insertAccountOwnerQuery mix ways to add arguments? I would have thought you could insert id and accountOwner.AccountID into the query with ExecContext, so I wanted to check if there's a limitation I'm missing since I haven't used it before.

If this is simply a style thing please disregard.